Userland/Utilities/pkg/AvailablePort.cpp

```diff
@@ -28,44 +28,56 @@
 #include <Shell/PosixParser.h>
 #include <Shell/Shell.h>
 
-static bool is_installed(HashMap<String, InstalledPort>& installed_ports_database, StringView package_name)
+void AvailablePort::query_details_for_package(HashMap<String, AvailablePort>& available_ports, HashMap<String, InstalledPort> const& installed_ports, StringView package_name, bool verbose)
 {
-    auto port = installed_ports_database.find(package_name);
-    return port != installed_ports_database.end();
-}
-
-static Optional<AvailablePort&> find_port_package(HashMap<String, AvailablePort>& available_ports, StringView package_name)
-{
-    auto port = available_ports.find(package_name);
-    if (port == available_ports.end())
-        return {};
-    return port->value;
-}
-
-ErrorOr<int> AvailablePort::query_details_for_package(HashMap<String, AvailablePort>& available_ports, HashMap<String, InstalledPort>& installed_ports, StringView package_name, bool verbose)
-{
-    auto possible_available_port = find_port_package(available_ports, package_name);
-    if (!possible_available_port.has_value()) {
+    auto possible_available_port = available_ports.find(package_name);
+    if (possible_available_port != available_ports.end()) {
```

Comment:
This check is wrong. `possible_available_port != available_ports.end()` is true if the port was found. You want `==` instead. That fixes the GCC issues for me too.

Reply:
Oh yes 🤦🏻